44-common-wallaroo/technical_specification.md

```diff
@@ -0,0 +1,39 @@
+# Choose frontend framework and component library (Common Wallaroo)
+
+**Epic Type:** Exploratory Epic
+
+## Principle Components of Exploration:
+
+In order to start refactoring our frontend, we need to make a final decision regarding the frontend framework, design system and  component library that we will use to build the next version of the data portal and potentially also the follow-up products.
+
+We are limiting the scope to the three most popular frameworks: React (Next.js), Angular, and Vue (Nuxt) that we have already used or evaluated in the Red Kangaroo Epic.
+
+This evaluation can be split into the following tasks:
+- Create a list of criteria for the frontend framework that are relevant for us (like: various core features, maturity, stability, popularity, future support, flexibility, completeness, static typing, documentation, testing, licensing, additional features like SSR).
+- Define weights for these criteria.
+- Evaluate the three frameworks given these criteria, using the insights gained in the Red Kangaroo epic, and finalize the decision for the frontend framework.
+- Find three responsive design systems with open source license that would work for us (e.g. Bootstrap, Google Material Design, Foundation). Consider that the design system should work particularly well on desktop screen sizes, mobile is less important for us.
+- Create a list of UI components that are most needed and important for us.
+- Create a list of 6 popular component libraries for the selected frontend framework supporting these components and design systems.
```

Comment:
Because in this case there are more than 3 obvious contenders, like NG Material, NG Bootstrap, NGX Bootstrap, NG Lightning, PrimeNG, Vaadin, Clarity, Onsen, Nebular, Syncfusion.

Picking only three would be a bit too arbitrary, I felt that picking 6, trying to rule out some of them upfront, was a good compromise.

Really evaluating a library upfront is difficult. They all look very shiny from outside, only when you are working with them for a longer time, you will detect the pain points. My idea is to not check everything ourselves but to rely on experience reports and assessments by others from reviews in the Internet - like if they all complain about bad support, we already have an estimate for that.